peer/brontide.go

```diff
@@ -2252,26 +2302,58 @@ func (p *Brontide) queueHandler() {
 // connection is still active.
 //
 // NOTE: This method MUST be run as a goroutine.
-func (p *Brontide) pingHandler(newPayload func() []byte) {
+func (p *Brontide) pingHandler(
+	newPayload func() []byte,
+	pongSize func() uint16,
+) {
+
 	defer p.wg.Done()
 
 	pingTicker := time.NewTicker(pingInterval)
 	defer pingTicker.Stop()
 
-	// TODO(roasbeef): make dynamic in order to create fake cover traffic
-	const numPongBytes = 16
-
 out:
 	for {
 		select {
 		case <-pingTicker.C:
 
+			pongBytes := pongSize()
 			pingMsg := &lnwire.Ping{
-				NumPongBytes: numPongBytes,
+				// TODO(roasbeef): make dynamic in order to
+				// create fake cover traffic
+				// NOTE(proofofkeags): this was changed to be
+				// dynamic to allow better pong identification,
+				// however, more thought is needed to make this
+				// actually usable as a traffic decoy
+				NumPongBytes: pongBytes,
 				PaddingBytes: newPayload(),
 			}
 
-			p.queueMsg(pingMsg, nil)
+			msgWritten := make(chan error)
+			p.queueMsg(pingMsg, msgWritten)
+			p.wg.Add(1)
+			go func() {
+				defer p.wg.Done()
+				err, ok := <-msgWritten
+				if err == nil && ok {
+					p.pingManager.markPingSend(pongBytes)
+					time.Sleep(pingTimeout)
```

Comment:
We can intead use a `time.AfterFunc` here: https://pkg.go.dev/time#AfterFunc

We'll still also retain the ability to cancel or reset the timer. It'll abstract out a bit from the goroutine logic here.